File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int colonIndex = logLine.IndexOf(':');
        string message = logLine.Substring(colonIndex+1).Trim();
        return message;
    }

    public static string LogLevel(string logLine)
    {
        int colonIndex = logLine.IndexOf(']');
        string error = logLine.Substring(1, colonIndex - 1).Trim().ToLower();
        return error;
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String message[] = logLine.split(": ");
        return message[1].trim();
    }

    public static String logLevel(String logLine) {
        String message[] = logLine.split(": ");
        String level = message[0].substring(1, message[0].indexOf(']'));
        return level.toLowerCase().trim();
    }

    public static String reformat(String logLine) {
        String last = logLevel(logLine);
        String first = message(logLine);
        return "%s (%s)".formatted(first, last);    
    }
}
